src/vast-client-serializer.js

```diff
@@ -0,0 +1,698 @@
+          }
+        })
+      }
+    }
+
+    if (mezzanine) {
+      mediaFilesObject = {
+        ...mediaFilesObject,
+        'Mezzanine': {
+          '@delivery': mezzanine.delivery,
+          '@type': mezzanine.type,
+          '@width': mezzanine.width,
+          '@height': mezzanine.height,
+          '@codec': mezzanine.codec,
+          '@id': mezzanine.id,
+          '@fileSize': mezzanine.fileSize,
+          '@mediaType': mezzanine.mediaType,
+          '#': this.wrapCDATA(mezzanine.fileURL),
+        }
+      }
+    }
+
+    if (interactiveCreativeFile?.fileURL) {
+      mediaFilesObject = {
+        ...mediaFilesObject,
+        'InteractiveCreativeFile': {
+          '@type': interactiveCreativeFile.type,
+          '@apiFramework': interactiveCreativeFile.apiFramework,
+          '@variableDuration': interactiveCreativeFile.variableDuration,
+          '#': this.wrapCDATA(interactiveCreativeFile.fileURL),
+        }
+      }
+    }
+
+    if (closedCaptionFiles?.length) {
+      mediaFilesObject = {
+        ...mediaFilesObject,
+        'ClosedCaptionFiles': {
+          'ClosedCaptionFile': closedCaptionFiles.map((closedCaptionFile) => {
+            return {
+              '@type': closedCaptionFile.type,
+              '@language': closedCaptionFile.language,
+              '#': this.wrapCDATA(closedCaptionFile.fileURL),
+            }
+          })
+        }
+      }
+    }
+
+    return mediaFilesObject
+  }
+
+  /**
+   * Build Icons node. IconClickFallbackImages children not supported.
+   * @param {Array[Object]} icons The list of icons objects
+   * @returns {Object} A Icons node
+   */
+  buildIcons(icons) {
+    if (!icons?.length) {
+      return null
+    }
+
+    return {
+      'Icon': icons.map((icon) => {
+        const staticResource = !icon.staticResource ? {} : {
+          'StaticResource': {
+            '@creativeType': icon.type,
+            '#': this.wrapCDATA(icon.staticResource),
+          },
+        }
+
+        return {
+          '@program': icon.program,
+          '@width': icon.width,
+          '@height': icon.height,
+          '@xPosition': icon.xPosition,
+          '@yPosition': icon.yPosition,
+          '@duration': this.convertToHHMMSS(icon.duration),
+          '@offset': icon.offset,
+          '@apiFramework': icon.apiFramework,
+          '@pxratio': icon.pxratio,
+          'IFrameResource': this.wrapCDATA(icon.iframeResource),
+          'HTMLResource': this.wrapCDATA(icon.htmlResource),
+          'IconClicks': {
+            'IconClickThrough': this.wrapCDATA(icon.iconClickThroughURLTemplate),
+            'IconClickTracking': icon.iconClickTrackingURLTemplates.map((iconClickTrackingURLTemplate) => {
+              return {
+                '@id': iconClickTrackingURLTemplate.id,
+                '#': this.wrapCDATA(iconClickTrackingURLTemplate.url), // TODO verify
+              }
+            }),
+            /* NonLinearClickTracking
```

Comment:
Is the commenting volontary ? :)

Reply:
It's also because it's not supported yet by the VAST-Client, I added a comment :)

Reply:
Perfect, thank you ! +1